src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -5471,7 +5471,7 @@ static void generate_fake_tag(const DoutPrefixProvider *dpp, RGWRados* store, ma
 
 static bool is_olh(map<string, bufferlist>& attrs)
 {
-  map<string, bufferlist>::iterator iter = attrs.find(RGW_ATTR_OLH_INFO);
+  map<string, bufferlist>::iterator iter = attrs.find(RGW_ATTR_OLH_VER);
```

Comment:
this part makes sense. `RGW_ATTR_OLH_VER` is set during `olh_init_modification()`, while `RGW_ATTR_OLH_INFO` is added later by `apply_olh_log()`. so for `get_obj_state()` calls in between, this prevents us from misinterpreting the olh head object as a normal head object. with this change, we'll go on to call `get_olh_target_state()` and `follow_olh()` to look up the current version. i just want to make sure that this call path returns `-ENOENT` until `apply_olh_log()` completes

currently, `follow_olh()` will return `-EINVAL` if it doesn't find the `RGW_ATTR_OLH_INFO` attr: https://github.com/ceph/ceph/blob/1b5e1bf/src/rgw/driver/rados/rgw_rados.cc#L7794-L7796

should that return `-ENOENT` now? there's a similar check in `RGWRados::get_olh()` that should probably change too

Reply:
I think that is a good point and sounds like you're probably correct. I'll review further to confirm as well.

Reply:
I did confirm this and updated it now